packages/airnode-node/src/types.ts

```diff
@@ -206,33 +214,39 @@ export interface AuthorizationByRequestId {
   readonly [requestId: string]: boolean;
 }
 
-export type ApiCallResponse = ApiCallSuccessResponse | ApiCallErrorResponse;
+export type RegularApiCallResponse = RegularApiCallSuccessResponse | ApiCallErrorResponse;
+export type HttpGatewayApiCallResponse = HttpGatewayApiCallSuccessResponse | ApiCallErrorResponse;
+export type HttpSignedDataApiCallResponse = HttpSignedDataApiCallSuccessResponse | ApiCallErrorResponse;
+
+export type ApiCallResponse = RegularApiCallResponse | HttpGatewayApiCallResponse | HttpSignedDataApiCallResponse;
+export interface RegularApiCallSuccessResponse {
+  success: true;
+  data: { encodedValue: string; signature: string };
+}
 
-export interface ApiCallSuccessResponse {
+export interface HttpGatewayApiCallSuccessResponse {
   success: true;
-  value: string;
-  signature: string;
+  data: { values: unknown[]; rawValue: unknown; encodedValue: string };
+}
+
+export interface HttpSignedDataApiCallSuccessResponse {
+  success: true;
+  // NOTE: "timestamp" and "encodedValue" are wrapped in data for legacy reasons
+  data: { data: { timestamp: string; encodedValue: string }; signature: string };
 }
```

Comment:
We already rely on that format in Airseeker, so I wanted to avoid unnecessary changes. An advantage of this PR is that the response value is better typed and there is no business logic in the deployer/handlers code.

Also, I am thinking whether we should include this into v0.6 release or wait and do it as part of v0.7, WDYT?

Reply:
Airseeker is still under heavy development and better to fix it now than later.

I want someone else to review it and properly test it so I don't think it will be ready for 0.6.

Reply:
We will do that anyway, I'd say change it.

Reply:
Implemented in https://github.com/api3dao/airnode/pull/871/commits/53049b16e12d1a1633d01f567a371331463ab71e